### ISSUE TITLE
Workaround for failed imports?

### ISSUE THREAD

**Issue by jelleag** [aspect-build/rules_js#574]
I am currently running into a variant of a known issue: ESM imports escape the runfiles tree and the sandbox due to [#362](https://github.com/aspect-build/rules_js/issues/362).

```
[vite]: Rollup failed to resolve import "@emotion/cache" from "../../../../../../../../node_modules/.aspect_rules_js/@mantine+styles@5.6.3_sogmqz4enknxtkxk3k5s6bqwnq/node_modules/@mantine/styles/esm/index.js".
This is most likely unintended because it can break your application at runtime.
If you do want to externalize this module explicitly add it to
`build.rollupOptions.external`
```

One workaround seems to be running the Vite binary from the root, but this is undesirable as I'm working with a monorepo. Are there currently any other workarounds, or maybe a timeline for the fix?

Many thanks in advance!

**Comment by gregmagolan** [comment]
#362 is a tricky one. Its a small change in the node source code that would fix it but the internal .js file that needs the change is compiled into the node binary so it would mean forking node and building it from source.

Your problem above however looks more like a pnpm strict issue to me.

It looks like @mantine/styles doesn't declare a dep on @emotion/cache: https://unpkg.com/@mantine/styles@5.6.3/package.json

You can inject the missing dependency with pnpm packageExtensions however. For example, we inject mocha-multi-reporters as a peer dep of mocha here: https://github.com/aspect-build/rules_js/blob/d1cb223b5cb43dcaa671aff2b6e84774f3d41617/package.json#L12

See https://pnpm.io/package_json#pnpmpackageextensions for more info

**Comment by jelleag** [comment]
@gregmagolan Greg, you are a hero. Have been struggling with this for so long. Your workaround works like a charm!

**Comment by gregmagolan** [comment]
Awesome.

BTW, I tried out the change in Node.js that is needed to fix #362 and verified that it does work as expected. https://github.com/aspect-build/rules_js/issues/362#issuecomment-1303912902. Next step is to get it landed in node.
